engine/global.gd

```diff
@@ -110,6 +110,7 @@ func change_map(map, entrance):
 	var new_map_path = "res://maps/" + map + ".tmx"
 	var new_map = load(new_map_path).instance()
 	
+	player.hud.update_buttons() # When changing zones, update the buttons shown
```

Comment:
What's the reason for calling `hud.update_buttons()` in the change map method? It seems a bit out of place.

Reply:
If controller is disconnected, It should update the buttons shown when changing zones

Reply:
Why when changing zones? Is there a way to make it event driven?

Reply:
I think i can do It connecting a Signal that checks when a controller is connected or disconnected